src/main/java/org/gitlab4j/api/SnippetsApi.java

```diff
@@ -0,0 +1,125 @@
+package org.gitlab4j.api;
+
+import java.util.List;
+import javax.ws.rs.core.GenericType;
+import javax.ws.rs.core.Response;
+import org.gitlab4j.api.models.Snippet;
+import org.gitlab4j.api.models.Visibility;
+
+/**
+ * This class provides an entry point to all the GitLab Snippets API project calls.
+ */
+public class SnippetsApi extends AbstractApi {
+
+	public SnippetsApi(GitLabApi gitLabApi) {
+		super(gitLabApi);
+	}
+
+    /**
+     * Create a new Snippet.
+     *
+     * @param title the title of the snippet
+     * @param fileName the file name of the snippet
+     * @param content the content of the snippet
+     * @return the created Snippet
+     * @throws GitLabApiException if any exception occurs
+     */
+	public Snippet createSnippet(String title, String fileName, String content) throws GitLabApiException {
+		GitLabApiForm formData = new GitLabApiForm()
+				.withParam("title", title)
+				.withParam("file_name", fileName)
+				.withParam("content", content);
+		
+		Response response = post(Response.Status.CREATED, formData, "snippets");
+		return (response.readEntity(Snippet.class));
+	}
+	
+    /**
+     * Create a new Snippet.
+     *
+     * @param title the title of the snippet
+     * @param fileName the file name of the snippet
+     * @param content the content of the snippet
+     * @param visibility the visibility (Public, Internal, Private) of the snippet
+     * @param description the description of the snippet
+     * @return the created Snippet
+     * @throws GitLabApiException if any exception occurs
+     */
+	public Snippet createSnippet(String title, String fileName, String content, Visibility visibility, String description) throws GitLabApiException {
+		GitLabApiForm formData = new GitLabApiForm()
+				.withParam("title", title)
+				.withParam("file_name", fileName)
+				.withParam("content", content)
+				.withParam("visibility", visibility)
+				.withParam("description", description);
+		
+		Response response = post(Response.Status.CREATED, formData, "snippets");
+		return (response.readEntity(Snippet.class));
+	}
+
+    /**
+     * Removes Snippet
+     *
+     * DELETE /snippets/:id
+     *
+     * @param snippetId the snippet ID to remove
+     * @throws GitLabApiException if any exception occurs
+     */
+	public void deleteSnippet(Integer snippetId) throws GitLabApiException {
+		if (snippetId == null) {
+			throw new RuntimeException("snippetId can't be null");
+		}
+		delete(Response.Status.NO_CONTENT, null, "snippets", snippetId);
+	}
+
+    /**
+     * Get a list of Authenticated User's Snippets.
+     *
+     * GET /snippets
+     *
+     * @return a list of authenticated user's snippets
+     * @throws GitLabApiException if any exception occurs
+     */
+	public List<Snippet> getSnippets() throws GitLabApiException {
```

Comment:
Add a parameter to this method indicating where to download the content, i.e.:
```
public List<Snippet> getSnippets(boolean downloadContent) throws GitLabApiException {
```
and the make fetching the content optional based on this.  You could add another method without the parameter that calls this method with ```downloadContent = false``` . The default should be false to make it act like the GitLab Snippets API.